src/gui/gui.cpp

```diff
@@ -4840,6 +4840,7 @@ bool FurnaceGUI::loop() {
     }
 
     if (displayNew) {
+      newSongQuery = "";
```

Comment:
Formatting: no spaces between operators.

Also, you can reset firstFrame of FurnaceGUI::drawNewSong() here. Add the variable to FurnaceGUI (src/gui/gui.h) and call it something like... newSongFirstFrame.